internal/cluster/validator.go

```diff
@@ -253,3 +253,33 @@ func (v *clusterValidator) printAllHostsAreReadyToInstall(context *clusterPrepro
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *clusterValidator) isDNSDomainDefined(c *clusterPreprocessContext) validationStatus {
+	return boolValue(c.cluster.BaseDNSDomain != "")
+}
+
+func (v *clusterValidator) printIsDNSDomainDefined(context *clusterPreprocessContext, status validationStatus) string {
+	switch status {
+	case ValidationFailure:
+		return "Base DNS Domain is undefined"
+	case ValidationSuccess:
+		return "Base DNS Domain is defined"
+	default:
+		return fmt.Sprintf("Unexpected status %s", status)
+	}
+}
+
+func (v *clusterValidator) isPullSecretSet(c *clusterPreprocessContext) validationStatus {
+	return boolValue(c.cluster.PullSecretSet)
```

Comment:
Is there a validation of pull secret? 
or is it ok if it fail only on kubeconfig generation?